python/GafferCyclesUI/CyclesOptionsUI.py

```diff
@@ -336,76 +236,81 @@ def __registerDevicePresets() :
-		)
+		presetNames.append( "{}/All".format( deviceType ) )
+		presetValues.append( "{}:*".format( deviceType ) )
 
-		Gaffer.Metadata.registerValue(
-			GafferCycles.CyclesOptions,
-			"options.device.value",
-			"preset:{}/All + CPU".format( deviceType ),
-			"CPU {}:*".format( deviceType )
-		)
+	Gaffer.Metadata.registerValue( "option:cycles:device", "presetNames", IECore.StringVectorData( presetNames ) )
+	Gaffer.Metadata.registerValue( "option:cycles:device", "presetValues", IECore.StringVectorData( presetValues ) )
 
 def __registerDenoiseDevicePresets() :
 
-	cpuRegistered = False
+	presetNames = [ "Automatic" ]
+	presetValues = [ "*" ]
+
 	typeIndices = {}
 	for device in GafferCycles.devices.values() :
 
 		# Ignore devices that don't support any denoisers
 		if device["denoisers"].value == 0 :
 			continue
 
-		if device["type"] == "CPU" :
-			if not cpuRegistered :
-				Gaffer.Metadata.registerValue( GafferCycles.CyclesOptions, "options.denoiseDevice.value", "preset:CPU", "CPU" )
-				cpuRegistered = True
+		if device["type"] == "CPU" and "CPU" not in presetNames :
+			presetNames.append( "CPU" )
+			presetValues.append( "CPU" )
 			continue
 
 		typeIndex = typeIndices.setdefault( device["type"], 0 )
 		typeIndices[device["type"]] += 1
 
-		Gaffer.Metadata.registerValue(
-			GafferCycles.CyclesOptions,
-			"options.denoiseDevice.value",
-			"preset:{}/{}".format( device["type"], device["description"] ),
-			"{}:{:02}".format( device["type"], typeIndex )
-		)
+		presetNames.append( "{}/{}".format( device["type"], device["description"] ) )
+		presetValues.append( "{}:{:02}".format( device["type"], typeIndex ) )
 
 	for deviceType, count in typeIndices.items() :
 
 		if count <= 1 :
 			continue
 
-		Gaffer.Metadata.registerValue(
-			GafferCycles.CyclesOptions,
-			"options.denoiseDevice.value",
-			"preset:{}/All".format( deviceType ),
-			"{}:*".format( deviceType )
-		)
+		presetNames.append( "{}/All".format( deviceType ) )
+		presetValues.append( "{}:*".format( deviceType ) )
+
+	Gaffer.Metadata.registerValue( "option:cycles:denoise_device", "presetNames", IECore.StringVectorData( presetNames ) )
+	Gaffer.Metadata.registerValue( "option:cycles:denoise_device", "presetValues", IECore.StringVectorData( presetValues ) )
+
+def __registerDenoiserPresets() :
+
+	presetNames = []
+	presetValues = []
+
+	if GafferCycles.hasOptixDenoise :
+		presetNames.append( "OptiX Denoiser" )
+		presetValues.append( "optix" )
+
+	if GafferCycles.hasOpenImageDenoise :
+		presetNames.append( "Open Image Denoise" )
+		presetValues.apend( "openimagedenoise" )
+
+	Gaffer.Metadata.registerValue( "option:cycles:integrator:denoiser_type", "presetNames", IECore.StringVectorData( presetNames ) )
+	Gaffer.Metadata.registerValue( "option:cycles:integrator:denoiser_type", "presetValues", IECore.StringVectorData( presetValues ) )
+
+def __registerPassPresets() :
+
+	presetNames = []
+	presetValues = []
+
+	for _pass in GafferCycles.passes.keys() :
```

Comment:
My script for comparing before/after on presets flagged a difference in ordering here. I think this is just because CompoundData ordering isn't stable between runs. Might be worth using `sorted( passes.keys() )`?

Reply:
Sorting the keys in e751c6c.